### PR TITLE
Add release drafter github action

### PR DESCRIPTION
**Pull request by ssbarnea**
This change should help preparing list of release notes for future
releases. When a maintainer will want to make a release he would
be able to tune the draft release notes and copy them to the
changelog.

The biggest advantage of this change is that the creation of the
draft is automated and it already uses the desired format, with
title link to PR and mention of author, reducing the maintenance
chore when preparing for a new release.

**Changelog-friendly one-liner**: 

##### Contributor checklist

- [x] Provided the tests for the changes.
- [x] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [x] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: .github/release-drafter.yml
Comment (atugushev) [inline .github/release-drafter.yml:1]: It looks like it's copied from Ansible, can we update the template in accordance with the current one. See [6.0.0](https://github.com/jazzband/pip-tools/releases/tag/6.0.0) release for example.

File: .github/workflows/labels.yml
Comment (atugushev) [inline .github/workflows/labels.yml:18]: These labels must be synced with the current ones.

**Review by atugushev — COMMENTED** [review]
Wow, that looks pretty useful! A few notes below:

File: .github/release-drafter.yml
Comment (webknjaz) [reply .github/release-drafter.yml:1]: From ansible-lint likely...

Comment (atugushev) [reply .github/release-drafter.yml:1]: Oh, [that](https://github.com/ansible-community/ansible-lint/releases) template looks pretty similar to ours.

**Comment by jezdez** [comment]
FTR, I've enabled allowing only GitHub released GitHub actions and verified actions from the Marketplace to make sure we don't have a wildgrowth of actions used. I'll be opening a discussion later in the members team to explain why I do that and how we'll have to batten down the hatches a bit.

**Comment by jezdez** [comment]
To be more clear, here's the set of actions allowed for this repo at the moment: 
CleanShot 2021-06-11 at 19 31 36@2x

**Comment by atugushev** [comment]
@jezdez FYI deadsnakes doesn't work and it breaks [CI](https://github.com/jazzband/pip-tools/actions/runs/931446852) workflow
image

**Comment by webknjaz** [comment]
@jezdez maybe those trailing commas break things?

**Comment by jezdez** [comment]
Oh, yeah, I fixed this already, this just needed “@*” appended.

File: .github/release-drafter.yml
Comment (ssbarnea) [reply .github/release-drafter.yml:1]: @atugushev Please just push all desired changes directly to that branch, it. would make it reach acceptable status much faster this way.

I found the list of currently configured labels bit confusing as I was not sure which ones end-up in which place in changelog. One thing that needs to be decided first is which is the one-of list. for sure we need one as `skip-changelog` for stuff that we do not want to mention in changelog bug for others we need a clean list.

The other label-check action does assure that we do not merge by mistake a change that is missing any-of the  required labels. Sadly there is no standardization in that area.

File: .github/workflows/release-drafter.yml
Comment (utkarsh2102) [inline .github/workflows/release-drafter.yml:15]: Good work, just a very trivial nit:

```suggestion
      # Drafts your next release notes as Pull Requests are merged into "master"
```

**Review by jezdez — CHANGES_REQUESTED** [review]
For the record, this needs to lose the use of the verify-pr-label-action action to be mergable. Release-drafter is already used in other Jazzband projects and is acceptable.

**Comment by ssbarnea** [comment]
@atugushev @jezdez Does it look ok now? Did I miss anything or is ready for approval. I can wait for two approvals for this one if you think that is needed.

**Comment by ssbarnea** [comment]
@atugushev Can you please review this? I would like to enable it so we can benefit from it for the new release.

File: .github/release-drafter.yml
Comment (atugushev) [inline .github/release-drafter.yml:50]: Trivial we use as "skip-changelog".

File: .github/release-drafter.yml
Comment (atugushev) [inline .github/release-drafter.yml:49]: The "maintenance" tag is not user-facing, so it could be excluded from "Other changes".

File: .github/release-drafter.yml
Comment (atugushev) [inline .github/release-drafter.yml:28]: Current template is:

```suggestion
change-template: "- $TITLE ([#$NUMBER](https://github.com/jazzband/pip-tools/pull/$NUMBER)). Thanks @$AUTHOR"
```

Comment (atugushev) [inline .github/release-drafter.yml:28]: If the change text comes from a PR title, then we should update the [pull request template](https://github.com/jazzband/pip-tools/blob/master/.github/PULL_REQUEST_TEMPLATE.md) accordingly.

**Review by atugushev — COMMENTED** [review]
Thanks for pinging! A few notes below:

**Comment by ssbarnea** [comment]
@atugushev Please give it another go, I hope I addressed all comments.

**Review by atugushev — APPROVED** [review]
LGTM 👍

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (webknjaz) [inline .github/PULL_REQUEST_TEMPLATE.md:6]:
```suggestion
- [ ] Assure PR title is short, clear, and good to be included in the user-oriented changelog
```

Comment (ssbarnea) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: @atugushev WTDY? Commit suggestion if you agree and lets move on.

File: .github/workflows/release-drafter.yml
Comment (jezdez) [inline .github/workflows/release-drafter.yml:13]: This is similar to how [prettytable has configured it](https://github.com/jazzband/prettytable/blob/5040d3b9e7d4f44cc901c8f44a5477fa57869632/.github/workflows/release-drafter.yml) and will restrict the permissions of the `GITHUB_TOKEN` a bit:

```suggestion
    permissions:
      contents: write
      pull-requests: read
    if: github.repository == 'jazzband/pip-tools'
    runs-on: ubuntu-latest
```

**Review by jezdez — CHANGES_REQUESTED** [review]
Tiny change to the action config needed, but otherwise this looks good to me.

**Comment by ssbarnea** [comment]
@jezdez Added that change. Please do not forget to approve it. As long you appear with "requested changes" we will not be able to merge the change, regardless other approvals.

**Review by jezdez — APPROVED** [review]

**Comment by jezdez** [comment]
> @jezdez Added that change. Please do not forget to approve it. As long you appear with "requested changes" we will not be able to merge the change, regardless other approvals.

Yep, that's how it was intended. Thanks for working on this @ssbarnea!

**Pull request merged**
